### PR TITLE
Add no-op variant of the API for unsupported browsers

### PR DESCRIPTION
**Pull request by iainmcgin**

### CODE REVIEW

File: ts/api/api.ts [hunk 1]
```diff
@@ -361,7 +361,73 @@ class InitializeOnDemandApi implements OnDemandOpenYoloApi {
...
 }
 
-export const openyolo: OnDemandOpenYoloApi = new InitializeOnDemandApi();
+/**
+ * Promise stub that immediately rejects.
+ */
+class FakePromise {
+  then(onFulfilled?: Function, onRejected?: Function): FakePromise {
+    window.setTimeout(() => {
+      onRejected('Unsupported browser');
```
Comment (TMSCH) [inline ts/api/api.ts]: Could that be an OpenYoloError instead of a string, to be consistent?

Comment (iainmcgin) [inline ts/api/api.ts]: Added OpenYoloError.unsupportedBrowser().

File: ts/api/api.ts [hunk 2]
```diff
@@ -361,7 +361,73 @@ class InitializeOnDemandApi implements OnDemandOpenYoloApi {
...
 }
 
-export const openyolo: OnDemandOpenYoloApi = new InitializeOnDemandApi();
+/**
+ * Promise stub that immediately rejects.
+ */
+class FakePromise {
+  then(onFulfilled?: Function, onRejected?: Function): FakePromise {
+    window.setTimeout(() => {
+      onRejected(OpenYoloError.unsupportedBrowser());
```
Comment (TMSCH) [inline ts/api/api.ts]: Oh I missed something here: if the onRejected function is not given, this will raise an error.

Comment (iainmcgin) [inline ts/api/api.ts]: Good catch, will fix.

**Comment by iainmcgin** [comment]
PTAL

**Comment by iainmcgin** [comment]
Testing it in-situ is difficult as it performs the replacement as a one-time operation on load of the module. I can refactor to at least pull out and test the logic that makes the decision.

**Comment by TMSCH** [comment]
Closing in favor of #61

**Pull request closed without merging**